Objects/descrobject.c

```diff
@@ -139,7 +141,7 @@ method_get(PyMethodDescrObject *descr, PyObject *obj, PyObject *type)
 {
     PyObject *res;
 
-    if (descr_check((PyDescrObject *)descr, obj, &res))
+    if (descr_check((PyDescrObject *)descr, obj, &res) > 0)
```

Comment:
I would prefer to leave the code as it is. In CPython, I saw two kinds of check:

* if (func()) ...  // something
* if (func() < 0) ... // error